src/lib/flexbox/api/layout-gap.ts

```diff
@@ -114,22 +115,25 @@ export class LayoutGapDirective extends BaseFxDirective implements AfterContentI
    * NOTE: this does NOT! differentiate between viewChildren and contentChildren
    */
   protected _watchContentChanges() {
-    let onMutationCallback = (mutations) => {
-      let validatedChanges = (it: MutationRecord) => {
-        return (it.addedNodes && it.addedNodes.length) ||
-            (it.removedNodes && it.removedNodes.length);
+    this._zone.runOutsideAngular(() => {
+
+      let onMutationCallback = (mutations) => {
+        let validatedChanges = (it: MutationRecord) => {
+          return (it.addedNodes && it.addedNodes.length) ||
+              (it.removedNodes && it.removedNodes.length);
+        };
+
+        // update gap styles only for child 'added' or 'removed' events
+        if (mutations.filter(validatedChanges).length) {
```

Comment:
Instead of using `mutation.filter(...).length`, you can use `mutations.some(...)` which will exit earlier.